### PR TITLE
Pin `psycopg-binary` dependency for easier dev env setup

### PR DESCRIPTION
**Pull request by sydney-runkle**
Closes https://github.com/pydantic/logfire/issues/490

Also, update deps with `uv sync --upgrade`, including a `pyright` bump hence the changes re type checking comments.

### CODE REVIEW

**Comment by sydney-runkle** [comment]
@Viicos, this might be a good one for you to take a look at - I got blocked with the `pyright` upgrade, and I know you're familiar with the recent changes...

File: tests/otel_integrations/test_psycopg.py [hunk 1]
```diff
@@ -23,7 +24,7 @@ def test_check_version():
 
 
 def test_instrument_psycopg():
-    original_connect = psycopg.connect
+    original_connect = cast('Callable[..., psycopg.Connection[Tuple[Any, ...]]]', psycopg.connect)
```
Comment (sydney-runkle) [inline tests/otel_integrations/test_psycopg.py]: This is probably wrong, I was just experimenting...

**Comment by Viicos** [comment]
I remember [this thread](https://github.com/psycopg/psycopg/issues/708) from the `psycopg` library. Until the latest version gets used, I'd suggest adding a `# pyright: reportUnknownMemberType=false` at the top of file.

**Comment by sydney-runkle** [comment]
> Until the latest version gets used

Ah ok, as in `psycopg[binary]` `latest`?

I think the pin might not actually be necessary here, waiting on a response to https://github.com/pydantic/logfire/issues/490#issuecomment-2414700487

**Comment by Viicos** [comment]
And unfortunately `psycopg` has implemented the generic `Connection` class in a way that is not compatible with `pyright` (and I believe the typing spec as well). So we may encounter more issues in the future

**Comment by sydney-runkle** [comment]
Ah ok, good to know. Thanks for the feedback @Viicos!

**Comment by sydney-runkle** [comment]
No longer needed, original issue closed.

**Pull request closed without merging**